### PR TITLE
tree: Cleanup and factor out APIs for emitting events

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Update/improve examples and docs for events.
Split up event implementation to put emitter logic in its own file.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: .changeset/smooth-yaks-unite.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+"@fluidframework/tree": minor
+---
+---
+"section": "tree"
+---
+Publish API for creating events as alpha.
+
+Export `createEmitter`, `EventEmitter` and related types as `@alpha`.
+See `EventEmitter`'s documentation for example usage.
```
Comment (CraigMacomber) [inline .changeset/smooth-yaks-unite.md:10]:
```suggestion
Export `createEmitterMinimal` and related types as `@alpha`.
See `createEmitterMinimal`'s documentation for example usage.
```

**Comment by CraigMacomber** [comment]
A usability issue with this APi has been reported:

Lack of strong typing of the keys least to string names for keys in code easily getting out of date (or having typos) with no compile errors.

There are two ways this could be improved:

- strong types on the keys in the set (ex: derive them from the object node or nodes provided)
- a way to refer to property key strings that is strongly typed to the object they are on and refactors properly. For example `Foo.fields.bar.key` possible leveraging a "FieldSchemaContextual" subclass which has the key as s strongly typed property.

File: packages/dds/tree/src/events/interop.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IEvent } from "@fluidframework/core-interfaces";
+import type { Listeners } from "./listeners.js";
+
+// TODO: this file is currently unused. Use it or remove it.
```
Comment (Josmithr) [inline packages/dds/tree/src/events/interop.ts:9]: For this PR? Or for the future?

File: packages/dds/tree/src/events/listeners.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * }>
+ * ```
+ * {@link createEmitter} can help implement this interface via delegation.
+ *
+ * @sealed @public
+ */
+export interface Listenable<TListeners extends object> {
+	/**
+	 * Register an event listener.
+	 * @param eventName - the name of the event
```
Comment (Josmithr) [inline packages/dds/tree/src/events/listeners.ts:52]: Nit (for here and other places where we aren't using `Complete sentence casing.`
```suggestion
	 * @param eventName - The name of the event.
```

**Review by Josmithr — APPROVED** [review]

File: packages/dds/tree/src/events/interop.ts (hunk 2 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/events/interop.ts:9]: Future. I assume this code has a reason it was written, but it does not document any, and isn't used, so I put it here and noted that.

I believe this is the logic to adapt tree's events to work with the framework/node event APIs.

**Pull request merged**